src/test/java/tests/TestUtils.java

```diff
@@ -77,4 +83,51 @@ public static void delay(double seconds) {
         }
     }
 
+    public static String readFileFromResource(Object callingObject, String filename) throws IOException {
+        ClassLoader classLoader = callingObject.getClass().getClassLoader();
+        return IOUtils.toString(classLoader.getResourceAsStream(filename));
+    }
+
+    /**
+     * Parses a string containing HTTP header fields and return a List of Header to be used with MockServer
+     * @param header
+     * @return
+     */
+    public static List<Header> parseHeaderRecord(String header) {
+        List<Header> result = new ArrayList<>();
+        String[] fields = header.split("[\r\n]+");
+
+        for (String field : fields) {
+            String[] nameAndValue = field.split(":", 2);
+            if (nameAndValue.length == 2) {
+                result.add(new Header(nameAndValue[0].trim(), nameAndValue[1].trim()));
+            }
+        }
+
+        return result;
+    }
+
+    /**
+     * Creates a GitHub API v3 request that can be used with MockServer and GitHubClient
+     * @param method the HTTP request method
+     * @param page the page of the resource being requested
+     * @param repoStringId string version of the repo id
+     * @param repoNumericId numeric version of the repo id returned in the Link header
+     * @param apiSegments segments of the request path after the repo id
+     * @return
+     */
+    public static HttpRequest createMockServerRequest(String method, String page,
```

Comment:
This is great, more high-level than I thought it would be.

Would it be better for the page and numeric id here to be ints, and only later converted to strings? This way we leverage the type system a bit more, people can't pass in random strings or null, etc.